worker/changestream/eventmultiplexer/eventmultiplexer.go

```diff
@@ -233,13 +252,17 @@ func (e *EventMultiplexer) loop() error {
 				continue
 			}
 
+			begin := e.clock.Now()
 			// Dispatch the set of changes, but do not cause the worker to
 			// exit. Just log out the error and then mark the term as done.
 			// There isn't anything we can do in this case.
 			if err := e.dispatchSet(changeSet); err != nil {
 				e.logger.Errorf("dispatching set: %v", err)
 				e.dispatchErrorCount++
+				// Increment number of dispatch errors metric.
+				e.metrics.DispatchErrorsInc()
 			}
+			e.metrics.DispatchDurationObserve(e.clock.Now().Sub(begin).Seconds())
```

Comment:
So we can do something fancy here. If we know there has been an error, we can pass it as `err == nil` if we're in an error state as a label. Then we'll know that the cause of a longer dispatch is because of the error case. As this is a labeled metric i.e. (WithLabelValues), we get this for free.

Reply:
done, it's a nice idea indeed